spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMergeOnReadUpdate.java

```diff
@@ -91,4 +155,19 @@ private void checkUpdateFileGranularity(DeleteGranularity deleteGranularity) {
             row(4, "it")),
         sql("SELECT * FROM %s ORDER BY dep ASC, id ASC", selectTarget()));
   }
+
+  private void initTable(String partitionedBy, DeleteGranularity deleteGranularity) {
```

Comment:
I understand we refactor common logic but this makes tests harder to read. For instance, the reader has no indication we add 4 batches in the init method.